shared-module/displayio/Shape.c

```diff
@@ -49,10 +49,16 @@ void common_hal_displayio_shape_construct(displayio_shape_t *self, uint32_t widt
     self->half_height = height;
 
     self->data = m_malloc(height * sizeof(uint32_t), false);
+
     for (uint16_t i = 0; i <= height; i++) {
         self->data[2 * i] = 0;
         self->data[2 * i + 1] = width;
     }
+
+    self->dirty_area.x1=0;
+    self->dirty_area.x2=width;
```

Comment:
I don't think you want any changes by default.

```suggestion
    self->dirty_area.x2=0;
```

Reply:
When the Shape is constructed, I think the dirty area should cover the whole Shape.  
I wrote it consistent with the constructor for `Bitmap.c` (see below).  Maybe I am missing something?

https://github.com/adafruit/circuitpython/blob/bbd802940ee707fbf8647e0f1017dfb713a94c14/shared-module/displayio/Bitmap.c#L67-L70

**Also, I feel like the `Shape` code has an "off by one" error in the for loop in line 53.**  The malloc is based on `height`, but the for loop here goes from `0` to `height` inclusive (which is `height+1` times).  I welcome your feedback.

Reply:
Ah ok. Yup, I think you are right about the off by one error.

Reply:
I fixed the off-by-one error and corrected an issue with `half_width` and `half_height` calculation.

Also, I clarified that the dirty rectangle tracking is exclusive for `.x2` and `.y2`.